### PR TITLE
test(playwright-ct): fix issues and flake

### PR DESCRIPTION
**Pull request by skogsmaskin**
### Description

I have identified several issues that caused the `playwright-ct` tests to be flaky.

Some were real issues, some were with the library itself, and some were issues with the tests (see individual commits for details).

I have upgraded the Playwright packages used in the mono-repo.

There might still be flake, but this should help at least.

This PR is also tied into https://github.com/sanity-io/sanity/pull/8021 and https://github.com/sanity-io/sanity/pull/8022 which was moved into own separate PRs

### What to review

* That the tests pass and have become less flaky.

### Testing

Should be tested automatically.

### Notes for release
N/A

### CODE REVIEW

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx [hunk 1]
```diff
@@ -44,34 +35,41 @@ export function PopoverEditDialog(props: PopoverEditDialogProps) {
...
         }
       },
-      [onClose],
+      [handleClose],
     ),
   )
 
   useClickOutsideEvent(
-    onClose,
+    () => handleClose,
```
Comment (skogsmaskin) [inline packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:72]: This can be shortened.

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx [hunk 2]
```diff
@@ -24,18 +25,8 @@ interface PopoverEditDialogProps {
...
-
-  // This hook is here to set open after the initial render.
-  // If rendered immediately, the popover will for a split second be
-  // visible in the top left of the boundaryElement before correctly
-  // placed pointing at the reference element.
-  useEffect(() => {
-    setOpen(true)
-  }, [])
-
   return (
```
Comment (skogsmaskin) [inline packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:30]: This is no longer needed after https://github.com/sanity-io/ui/commit/c95c1c014a311747286491781db0bd630f43d63b

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx [hunk 3]
```diff
@@ -80,42 +78,44 @@ function Content(props: PopoverEditDialogProps) {
   const [contentElement, setContentElement] = useState<HTMLDivElement | null>(null)
   const containerElement = useRef<HTMLDivElement | null>(null)
 
-  useEffect(() => {
-    // When rendered, focus on the first input element in the content
-    if (contentElement) {
-      contentElement.querySelector('input')?.focus()
-    }
-  }, [contentElement])
```
Comment (skogsmaskin) [inline packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:88]: This should rather be handled by the `autoFocus` prop so that it works like the other Modals we use (where the first focusable element is focused, usually the close-button).

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx [hunk 4]
```diff
@@ -80,42 +78,44 @@ function Content(props: PopoverEditDialogProps) {
   const [contentElement, setContentElement] = useState<HTMLDivElement | null>(null)
   const containerElement = useRef<HTMLDivElement | null>(null)
 
-  useEffect(() => {
```
Comment (skogsmaskin) [inline packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:83]: This should rather be handled by the autoFocus prop so that it works like the other Modals we use (where the first focusable element is focused, usually the close-button).

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx (hunk 3 above)
Comment (skogsmaskin) [inline packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:88]: This should rather be handled by the autoFocus prop so that it works like the other Modals we use (where the first focusable element is focused, usually the close-button).

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx [hunk 5]
```diff
@@ -80,42 +78,44 @@ function Content(props: PopoverEditDialogProps) {
...
       scrollElement={contentElement}
       containerElement={containerElement}
     >
-      <Flex ref={containerElement} direction="column" height="fill">
-        <ContentHeaderBox flex="none" padding={1}>
-          <Flex align="center">
-            <Box flex={1} padding={2}>
-              <Text weight="medium">{title}</Text>
-            </Box>
+      <FocusLock autoFocus whiteList={handleFocusLockWhiteList}>
```
[the remark this answers is not in the document]
Comment (bjoerge) [reply packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:93]: This is great! However, it doesn't seem to reflect the title of the PR which implies the changes only affects tests. Could we either get it split into two PRs or change the title?

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx (hunk 2 above)
Comment (bjoerge) [reply packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:30]: Nice!

File: packages/sanity/package.json [hunk 6]
```diff
@@ -266,8 +266,8 @@
     "yargs": "^17.3.0"
   },
   "devDependencies": {
-    "@playwright/experimental-ct-react": "1.44.1",
-    "@playwright/test": "1.44.1",
+    "@playwright/experimental-ct-react": "1.49.1",
```
Comment (bjoerge) [inline packages/sanity/package.json:269]: wonder when it will no longer be experimental 🤔

File: packages/sanity/playwright-ct.config.ts [hunk 7]
```diff
@@ -92,6 +92,7 @@ export default defineConfig({
           firefoxUserPrefs: {
             'dom.events.asyncClipboard.readText': true,
             'dom.events.testing.asyncClipboard': true,
+            'ui.prefersReducedMotion': 1,
```
Comment (bjoerge) [inline packages/sanity/playwright-ct.config.ts:95]: why was this added? (would be great to have a comment)

File: packages/sanity/package.json (hunk 6 above)
Comment (skogsmaskin) [reply packages/sanity/package.json:269]: Haha, yes

File: packages/sanity/playwright-ct.config.ts (hunk 7 above)
Comment (skogsmaskin) [reply packages/sanity/playwright-ct.config.ts:95]: The idea is to speed up the tests by reducing the CSS animations (less CPU time used).

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (skogsmaskin) [reply packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx:93]: They are a bit tied together, but yes. I can split this one out into a separate PR that can be merged into this one?

[the remark this answers is not in the document]
Comment (skogsmaskin) [reply packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx]: Split into https://github.com/sanity-io/sanity/pull/8022 and https://github.com/sanity-io/sanity/pull/8022

File: packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx (hunk 3 above)
Comment (skogsmaskin) [reply packages/sanity/src/core/form/inputs/PortableText/object/modals/PopoverModal.tsx]: No longer relevant here, moved to https://github.com/sanity-io/sanity/pull/8021

**Review by bjoerge — APPROVED** [review]
This is great, thank you so much!
